torchvision/prototype/transforms/_container.py

```diff
@@ -21,16 +19,21 @@ def forward(self, *inputs: Any) -> Any:
         return sample
 
 
-class RandomApply(_RandomApplyTransform):
-    def __init__(self, transform: Transform, *, p: float = 0.5) -> None:
-        super().__init__(p=p)
-        self.transform = transform
+class RandomApply(Compose):
```

Comment:
I don't really understand why `RandomApply` needs to support multiple transforms. I've re-added it here for BC, but I think we should deprecate this in favor of what our internal `_RandomApplyTransform` does. If the user needs multiple transforms, they can simply wrap them in a `Compose` before passing it to `RandomApply`.

Reply:
I agree. Maybe, we can extend `Compose` with `p` argument to cover `RandomApply` feature...

Reply:
I think for now we should keep it and review on the future what we want to deprecate. It's true that many of the transforms can be written as a combination of other transforms. For example, the majority of the `Random*` transforms that support a `p` probability could have used the `RandomApply`. Unfortunately dropping some of these functionalities not only breaks BC but also leads to more verbose code. So we should have a separate discussion over what should be deprecated and why.